src/Okta.Idx.Sdk.UnitTests/IdxClientLayer1Should.cs

```diff
@@ -0,0 +1,2711 @@
+                            {
+                                ""name"": ""stateHandle"",
+                                ""required"": true,
+                                ""value"": ""02eYb3-CKPkh3l9SMxym9zyPZt1bLo5q_yab6ahhFg"",
+                                ""visible"": false,
+                                ""mutable"": false
+                            }
+                        ],
+                        ""accepts"": ""application/ion+json; okta-version=1.0.0""
+                    },
+                    ""app"": {
+                        ""type"": ""object"",
+                        ""value"": {
+                            ""name"": ""okta_enduser"",
+                            ""label"": ""okta_enduser"",
+                            ""id"": ""DEFAULT_APP""
+                        }
+                    }
+                }
+            ";
+
+            #endregion
+
+            var mockRequestExecutor = new MockedStringRequestExecutor(rawSendEmailCodeResponse);
+            var testClient = new TesteableIdxClient(mockRequestExecutor);
+
+            // Create a mock for an introspect response with "Identify" as a remediation option
+            var resourceFactory = new ResourceFactory(
+                testClient,
+                NullLogger.Instance,
+                new AbstractResourceTypeResolverFactory(ResourceTypeHelper.GetAllDefinedTypes(typeof(Resource))));
+            var data = new DefaultSerializer().Deserialize(rawChallengeEmailResponse);
+            var challengeResponse = resourceFactory.CreateNew<IdxResponse>(data);
+
+
+            challengeResponse.Remediation.RemediationOptions.FirstOrDefault().Href.Should()
+                .Be("https://foo.okta.com/idp/idx/challenge/answer");
+            var identifyRemediationOption = challengeResponse.Remediation.RemediationOptions.FirstOrDefault();
+
+            // Create a mock for a possible body when making a call to /identify
+            var remediationProceedRequest = new IdxRequestPayload()
+                                                {
+                                                    StateHandle = "02eYb3-CKPkh3l9SMxym9zyPZt1bLo5q_yab6ahhFg",
+                                                };
+
+            remediationProceedRequest.SetProperty("credentials", new { passcode = "31416", });
+
+            var identifyProcessedResponse = await identifyRemediationOption.ProceedAsync(remediationProceedRequest);
+
+            identifyProcessedResponse.Should().NotBeNull();
+            identifyProcessedResponse.StateHandle.Should().NotBeNullOrEmpty();
+            identifyProcessedResponse.Version.Should().NotBeNullOrEmpty();
+            identifyProcessedResponse.ExpiresAt.Value.Should().Be(DateTimeOffset.Parse("2020-10-22T18:35:22.000Z"));
+            identifyProcessedResponse.Intent.Should().Be("LOGIN");
+
+            // TODO: Revisit this is not null now
+            //identifyProcessedResponse.Remediation.Should().BeNull();
```

Comment:
Is this of concern?

Reply:
Yes, there's an open jira that needs to be addressed.